### ISSUE TITLE
`S3FileSystem.du()` FileNotFoundError for paths with empty prefixes

### ISSUE THREAD

**Issue by andyfarmerboy** [fsspec/s3fs#450]
The `.du()` method returns a FileNotFoundError if any terminal prefix trees at or below the path root do not contain any files.

```python
path = 'bucket/empty-prefix/'  
print(S3FileSystem.du(path))

#> FileNotFoundError

# (Expected) 
# 0
```

The method returns a numeric value as expected if all prefixes contain files.

**Comment by martindurant** [comment]
Agreed that this should work. See the general discussion at https://github.com/intake/filesystem_spec/issues/562 - we could do with a comprehensive set of tests for the object storage backends with paths with/without terminal "/", and with/without deeper-nested contents, across the several methods that do this kind of work.

For now, there is a chance that this particular issue is already fixed on main (soon to be released), because `find`, used by `du`, no longer uses the prefix/delimiter approach when maxdepth is not specified.

**Comment by ianthomas23** [comment]
I have confirmed that this has already been fixed, and I've submitted a PR (#803) to ensure there are no future regressions.
